src/only-one/only-one.ts

```diff
@@ -25,10 +29,11 @@ const monkey = new Monkey()
 const snake = new Snake()
 
 type AnimalInZoo = Monkey | Snake
-
+//Animal shoul implement behavior - eat
 function eatSomething(animal: AnimalInZoo) {
-    animal.eatBanana();
-    animal.eatMouse();
+
+   return animal.eat();
```

Comment:
But you break the rules here ;) Do not modify those classes.

Reply:
What about liskov substitution principle ;)  It will be necessary to check the type of AnimalInZoo, what can be ugly. 

`const monkey = new Monkey() as AnimalInZoo
const snake = new Snake() as AnimalInZoo

type AnimalInZoo = Monkey & Snake
function eatSomething(animal: AnimalInZoo) {
    return animal.eatBanana?animal.eatBanana():animal.eatMouse();
}
`